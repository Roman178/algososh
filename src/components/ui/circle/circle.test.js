import { Circle } from "./circle";
import renderer from "react-test-renderer";
import { Button } from "../button/button";

describe("Circle renders correctly", () => {
  it("without letters", () => {
    const CircleWithouText = renderer.create(<Circle />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with letters", () => {
    const CircleWithouText = renderer.create(<Circle letter="abcd" />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with head", () => {
    const CircleWithouText = renderer.create(<Circle head="head" />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with head as the React component", () => {
    const CircleWithouText = renderer.create(<Circle head={<Button />} />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with tail", () => {
    const CircleWithouText = renderer.create(<Circle tail="head" />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with tail as the React component", () => {
    const CircleWithouText = renderer.create(<Circle tail={<Button />} />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("with index", () => {
    const CircleWithouText = renderer.create(<Circle index={1} />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("when isSmall prop is true", () => {
    const CircleWithouText = renderer.create(<Circle isSmall={true} />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("when default state", () => {
    const CircleWithouText = renderer.create(<Circle state="default" />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("when changing state", () => {
    const CircleWithouText = renderer.create(<Circle state="changing" />);
    expect(CircleWithouText).toMatchSnapshot();
  });

  it("when modified state", () => {
    const CircleWithouText = renderer.create(<Circle state="modified" />);
    expect(CircleWithouText).toMatchSnapshot();
  });
});
