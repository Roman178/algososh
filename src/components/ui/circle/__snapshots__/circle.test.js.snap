// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle renders correctly when changing state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly when default state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly when isSmall prop is true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly when modified state 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with head as the React component 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      />
    </button>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle renders correctly with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      abcd
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle renders correctly with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    head
  </div>
</div>
`;

exports[`Circle renders correctly with tail as the React component 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <button
      className="text text_type_button text_color_primary button undefined false "
      type="button"
    >
      <p
        className="text undefined"
      />
    </button>
  </div>
</div>
`;

exports[`Circle renders correctly without letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
